lib/usb-device.js

```diff
@@ -0,0 +1,238 @@
+// import constants
+import { USB_VENDOR_IDS, ADB_VALUES, CONNECT_VALUES,
+          ADB_COMMANDS, ADB_KEY } from './constants';
+let MAXDATA = CONNECT_VALUES.CONNECT_MAXDATA;
+// import helpers
+import { generateMessage } from './helpers';
+import { packetFromBuffer } from './helpers';
+
+// required libraries
+import path from 'path';
+import fs from 'fs';
+import Promise from 'bluebird';
+import * as usb from 'usb';
+Promise.promisifyAll(usb);
+
+// local constants
+const homedir = (process.platform === 'win32') ? process.env.HOMEPATH
+                  : process.env.HOME;
+const keyPath = path.join(homedir, ADB_KEY.PUBLIC_KEY);
+const publicKeyString = fs.readFileSync(keyPath);
```

Comment:
Okay, what about the `readFile`? I'm not sure what the best way is to export that value if we want to use the async version of the function as well as make sure the `readFile` returned something.  We can't even complete the handshake with a device if we can't get the adb keys.  I feel like using export and await together is a bad thing to do?